cmds/ls/fileinfo_linux.go

```diff
@@ -34,10 +34,21 @@ type fileInfo struct {
 	uid, gid     uint32
 	size         int64
 	modTime      time.Time
+	symlink      string
 }
 
-func extractImportantParts(fi os.FileInfo) fileInfo {
+func extractImportantParts(n string, fi os.FileInfo) fileInfo {
+	var link string
+
 	s := fi.Sys().(*syscall.Stat_t)
+	if fi.Mode()&os.ModeType == os.ModeSymlink {
+		if l, err := os.Readlink(n); err != nil {
+			link = err.Error()
```

Comment:
Yeah I thought about this but this function doesn't have an error return. I went with the simplest approach, though it might suck in the long run. I have an immediate need for this to work :-)